StandardLibrary/src/String.dfy

```diff
@@ -48,4 +50,26 @@ module StandardLibrary.String {
   {
     Int2String(n, Base10)
   }
+
+  /* Returns the index of a substraing or None, if the substring is not in the string */
+  method HasSubString(xs: string, ss: string)
+    returns (o: Wrappers.Option<nat>)
+    requires |ss| <= |xs|
+    ensures o.Some? ==>
+              o.value <= |xs| - |ss| && xs[o.value..(o.value + |ss|)] == ss
+    // TODO: ensures o.None? ==> no such index exists
+  {
+    var index: nat := 0;
+    var limit: nat := |xs| - |ss|;
+    while index <= limit
+      decreases limit - index
+    {
+      if (xs[index .. index + |ss|] == ss) {
```

Comment:
FYI - this is an allocation of a sequence of size |ss|

Reply:
Is there a non-allocation comparison?